### ISSUE TITLE
SimpleDiscoveryClient does not publish an InstanceRegisteredEvent?

### ISSUE THREAD

**Issue by gdubya** [spring-cloud/spring-cloud-commons#410]
We are developing a collection of services to deploy into an Openshift environment. One of the applications is a simple Spring Cloud Gateway instance. 

When developing a service locally (e.g. _serviceA_) we would like to also test the gateway to access _serviceA_. We are using the SimpleDiscoveryClient in the Gateway to configure the location of _serviceA_. We noticed that the SimpleDiscoveryClientAutoConfiguration does not publish a InstanceRegisteredEvent, which means that the DiscoveryClientHealthIndicator is never initialized.  Is this intentional?

Our Gateway pom depends on spring-cloud-dependencies Finchley.SR1, which depends on spring-cloud-commons 2.0.1.RELEASE.

**Comment by ryanjbaxter** [comment]
I believe it is a `ServiceRegistry` that would fire a `InstanceRegisteredEvent`.  I dont see how using a `DiscoveryClient` would cause that to happen.

**Comment by gdubya** [comment]
That makes sense. But in this case, with SimpleDiscoveryClient, we're not using a service registry. In the (deprecated) NoopDiscoveryClientAutoConfiguration there is a method for publishing the event when the context is refreshed.

**Comment by ryanjbaxter** [comment]
Yeah I guess we are technically registering a service(s) in `SimpleDiscoveryClientAutoConfiguration`.  @spencergibb what do you think?
